source/Irrlicht/COpenGLDriver.h

```diff
@@ -946,9 +959,7 @@ namespace video
             };
 
             //!
-            typedef std::pair<COpenGLVAOSpec::HashAttribs,COpenGLVAO*> HashVAOPair;
-            HashVAOPair                 CurrentVAO;
-            core::vector<HashVAOPair>    VAOMap;
+            core::vector<SOpenGLState::HashVAOPair> VAOMap;

```

Comment:
why is it s veector and not a map?

Reply:
its how it was

Reply:
any reason not to improve on it?

Reply:
dont you think vector is best here? (allocates much less often)

Reply:
the bottleneck is not cache population, but fast retrieval.

Reply:
So its binary insertion sorted vector-based cache, vs map<> based cache

If cache will be large enough not to LRU evict any VAOs in a frame on average scenes, then vector can stay.

But other resources wil definitely need a map<>

Reply:
@Crisspl status on other resource caches?

Reply:
pipeline cache partially done